frontend/app/src/components/history/TransactionEventForm.vue

```diff
@@ -223,326 +224,280 @@ import { bigNumberifyFromRef, One, Zero } from '@/utils/bignumbers';
-      type: Function as PropType<
-        (
-          event: NewEthTransactionEvent | EthTransactionEventEntry
-        ) => Promise<ActionStatus>
-      >
-    }
+const props = defineProps({
+  value: { required: false, type: Boolean, default: false },
+  edit: {
+    required: false,
+    type: Object as PropType<EthTransactionEvent>,
+    default: null
   },
-  emits: ['input'],
-  setup(props, { emit }) {
-    const { t } = useI18n();
-    const { edit, transaction, saveData } = toRefs(props);
+  transaction: {
+    required: false,
+    type: Object as PropType<EthTransactionEntry>,
+    default: null
+  },
+  saveData: {
+    required: true,
+    type: Function as PropType<
+      (
+        event: NewEthTransactionEvent | EthTransactionEventEntry
+      ) => Promise<ActionStatus>
+    >
+  }
+});
 
-    const input = (valid: boolean) => emit('input', valid);
+const emit = defineEmits(['input']);
+const { t } = useI18n();
+const { edit, transaction, saveData } = toRefs(props);
+
+const input = (valid: boolean) => emit('input', valid);
+
+const { isTaskRunning } = useTasks();
+const { currencySymbol } = storeToRefs(useGeneralSettingsStore());
+const { exchangeRate, getHistoricPrice } = useBalancePricesStore();
+
+const lastLocation = useLocalStorage(
+  'rotki.ledger_action.location',
+  TRADE_LOCATION_EXTERNAL
+);
+
+const identifier = ref<number | null>(null);
+const eventIdentifier = ref<string>('');
+const sequenceIndex = ref<string>('');
+const datetime = ref<string>('');
+const location = ref<string>('');
+const eventType = ref<string>('');
+const eventSubtype = ref<string>('');
+const transactionEventType = ref<string | null>();
+const asset = ref<string>('');
+const amount = ref<string>('');
+const fiatValue = ref<string>('');
+const locationLabel = ref<string>('');
+const notes = ref<string>('');
+const counterparty = ref<string>('');
+
+const rate = ref<string>('');
+const errorMessages = ref<{ [field: string]: string[] }>({});
+
+const locationRules = [
+  (v: string) =>
+    !!v ||
+    t('transactions.events.form.location.validation.non_empty').toString()
+];
+
+const assetRules = [
+  (v: string) =>
+    !!v || t('transactions.events.form.asset.validation.non_empty').toString()
+];
+
+const amountRules = [
+  (v: string) =>
+    !!v || t('transactions.events.form.amount.validation.non_empty').toString()
+];
+
+const fiatValueRules = [
+  (v: string) =>
+    !!v ||
+    t('transactions.events.form.fiat_value.validation.non_empty', {
+      currency: get(currencySymbol)
+    }).toString()
+];
+
+const sequenceIndexRules = [
+  (v: string) =>
+    !!v ||
+    t('transactions.events.form.sequence_index.validation.non_empty').toString()
+];
+
+const eventTypeRules = [
+  (v: string) =>
+    !!v ||
+    t('transactions.events.form.event_type.validation.non_empty').toString()
+];
+
+const eventSubtypeRules = [
+  (v: string) =>
+    !!v ||
+    t('transactions.events.form.event_subtype.validation.non_empty').toString()
+];
```

Comment:
Since you are already migrating the component maybe also do the vuelidate treatment?